### ISSUE TITLE
Move Favorites From .godot to Editor/Project Settings Directory

### ISSUE THREAD

**Issue by karomapu** [godotengine/godot-proposals#10746]
### Describe the project you are working on

The proposal is for the editor and not project-specific.

### Describe the problem or limitation you are having in your project

If the user needs to remove the `.godot` folder for any reason (say, to force reimport of the whole project) that also means favorites are removed along with it. This can be a huge issue in the middle of development where the user is reliant on favorites to quickly navigate around the file system.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It will prevent favorites from being carelessly deleted from the project. Since it is something that is created and maintained by the user, and not auto-generated by Godot, it doesn't belong in the `.godot` directory.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

As stated in the title, store favorites in the Editor Data/Settings folder instead. It seems some project-specific settings are also already stored there, why not favorites too?

Alternatively, maybe give users the option to save it in project settings if they want it to be shared across multiple team members/machines.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Sure, but this feature really should be available out of the box.

### Is there a reason why this should be core and not an add-on in the asset library?

The favorites feature is built-in to godot. I don't see why this needs to be an add-on to improve core functionality.

**Comment by tetrapod00** [comment]
Wherever this is stored, it should be a per-user setting, not a per-project setting, right? Otherwise you'll run into the problem of multiple team members overwriting each other's favorites, and adding noise to version control history.

**Comment by karomapu** [comment]
right, I initially thought that favorites could be project-wide and wouldn't need to be updated often, but now I realize individual members would still want their own user-defined favorites and kinda defeats the purpose. Maybe users can override the project's favorites locally, but that could make things messy

**Comment by Calinou** [comment]
If this is to be stored in the editor settings (which I think makes the most sense), it should still be scoped on a per-project basis so that identical nodes or file paths in multiple projects aren't considered favorites in all projects. There is a way to define per-project metadata that is still stored in the [editor settings folder](https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html#editor-data-paths): `EditorSettings::get_singleton()->get_project_metadata()`.

**Comment by actforjason** [comment]
> If this is to be stored in the editor settings (which I think makes the most sense), it should still be scoped on a per-project basis so that identical nodes or file paths in multiple projects aren't considered favorites in all projects. There is a way to define per-project metadata that is still stored in the [editor settings folder](https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html#editor-data-paths): `EditorSettings::get_singleton()->get_project_metadata()`.

Maybe have both global and local favorites #14379 .

**Comment by KoBeWi** [comment]
> If the user needs to remove the .godot folder for any reason

With https://github.com/godotengine/godot/pull/120093 merged, you no longer need to do that ever. The Clear Cache option removes relevant files, while keeping user settings intact.

> Since it is something that is created and maintained by the user, and not auto-generated by Godot, it doesn't belong in the .godot directory.

Various toggles and editor states (like per-scene grid settings or node folding) are also stored in the `.godot` folder. Should they be moved out too?
